src/core/meshcop/meshcop_tlvs.hpp

```diff
@@ -971,52 +971,31 @@ class SecurityPolicyTlv : public Tlv, public TlvInfo<Tlv::kSecurityPolicy>
      * @retval FALSE  If the TLV does not appear to be well-formed.
      *
      */
```

Comment:
I think with the flag extension the  `SecurityPolicyTlv` can have different overall TLV length?
- When appending it to messages, do we always use the extended model (largest TLV length)? Otherwise, we need to be be careful when appending/writing this TLV to a message (some of the `Tlv` methods use the overall size of `Tlv` type).

Reply:
Agree. I think a Thread 1.1 device should always append `SecurityPolicyTlv` with 1B flags, and 1.2 device 2B flags.
The best way to guarantee that is to define different structure for different Thread version.